### PR TITLE
fix(SideBarPopup): close sidebarPopup on ESC keypress

### PR DESCRIPTION
**Pull request by nileshgulia1**
As an extension of https://github.com/plone/volto/pull/2509#issuecomment-1016368522
The problem with block losing focus on ESC was originating from https://github.com/plone/volto/blob/355bbad4e61420c4ede481a1bcae21a3ed82148e/packages/volto/src/components/manage/Blocks/Block/BlocksForm.jsx#L77
The solution was to intercept the event by making `<aside>` focusable. Since non-interactive elements should not be made focusable, therefore added tabIndex={-1}.

### CODE REVIEW

**Comment by nileshgulia1** [comment]
Edit: I think introducing focus was a smell, I modified using intercepting even conditionally.

**Comment by sneridagh** [comment]
@nileshgulia1 this has to be merged along with the other?
From the comments in the other PR I understand that it's solved already and no longer reproduceable.

**Review by sneridagh — APPROVED** [review]

File: packages/volto/src/components/manage/Sidebar/SidebarPopup.jsx [hunk 1]
```diff
@@ -16,18 +16,26 @@ const SidebarPopup = (props) => {
...
+  };
+
   const [isClient, setIsClient] = React.useState(false);
   React.useEffect(() => {
     setIsClient(true);
   }, []);
 
   React.useEffect(() => {
     document.addEventListener('mousedown', handleClickOutside, false);
+    document.addEventListener('keyup', handleEscapeKey, false);
```
Comment (sneridagh) [inline packages/volto/src/components/manage/Sidebar/SidebarPopup.jsx:33]: Just a comment why adding the `keyup`?

Comment (sneridagh) [reply packages/volto/src/components/manage/Sidebar/SidebarPopup.jsx:33]: @nileshgulia1 ^^

Comment (nileshgulia1) [reply packages/volto/src/components/manage/Sidebar/SidebarPopup.jsx:33]: @sneridagh I had to cut down the propagation to https://github.com/plone/volto/blob/1b8328cc0029f092525f6791d5458665b5008a4b/packages/volto/src/helpers/Utils/useDetectClickOutside.js#L52 that removes the focus on selected block.

**Pull request merged**